### PR TITLE
Late minor fixes for --format=LIST

### PR DESCRIPTION
**Pull request by magnumripper**
- Move all handling of [Disabled:Formats] to our new functions.
- Drop obsolete composite group dynamic-all. New syntax would be, for
  example, +dynamic,all.
- Do not complain about an inclusion not matching if an earlier inclusion
  added a superset (eg. --format=raw*,raw-md5).
- Update doc/OPTIONS and split it into paragraphs.

Still not breaking `make generic` ;-)

### CODE REVIEW

File: doc/OPTIONS
Comment (magnumripper) [inline doc/OPTIONS:398]: As the section for --format is so large, I split it into paragraphs. For separating *that* from the rest, I added empty lines (so they are now double empty lines) above and below the whole section.

Comment (solardiz) [reply doc/OPTIONS:398]: Perhaps we should have empty lines between descriptions of _every_ option, then - for consistency.

Comment (magnumripper) [reply doc/OPTIONS:398]: Yes, and actually we should review the whole file in case more sections should be split the same way. Also, the used line width varies a little, between below 70 to at least a full 80, which is not so good. Oh, and we're not consistently using double space after period.

That'd be a separate task though.

**Review by ghost — APPROVED** [review]
Do you need an approval for merging?

**Comment by magnumripper** [comment]
> Do you need an approval for merging?

Yes, thanks. I'll merge this as it fixes new bugs and not likely introduces new ones

**Pull request merged**
